cmd/soroban-cli/src/commands/network/config_settings.rs

```diff
@@ -0,0 +1,86 @@
+use crate::config::network;
+use crate::print::Print;
+use crate::{commands::global, config};
+use clap::command;
+use semver::Version;
+use stellar_xdr::curr::{
+    ConfigSettingId, ConfigUpgradeSet, LedgerEntryData, LedgerKey, LedgerKeyConfigSetting,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Config(#[from] config::Error),
+    #[error(transparent)]
+    Network(#[from] network::Error),
+    #[error(transparent)]
+    Serde(#[from] serde_json::Error),
+    #[error(transparent)]
+    Rpc(#[from] soroban_rpc::Error),
+    #[error(transparent)]
+    Semver(#[from] semver::Error),
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, clap::ValueEnum, Default)]
+pub enum OutputFormat {
+    /// JSON result of the RPC request
+    #[default]
+    Json,
+    /// Formatted (multiline) JSON output of the RPC request
+    JsonFormatted,
+}
+
+#[derive(Debug, clap::Parser, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub config: config::ArgsLocatorAndNetwork,
+    /// Format of the output
+    #[arg(long, default_value = "json")]
+    pub output: OutputFormat,
+}
+
+impl Cmd {
+    pub async fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let rpc = self.config.get_network()?.rpc_client()?;
+
+        // If the network protocol version is ahead of the XDR version (which tracks the protocol
+        // version), there could be config settings defined in the newer protocol version that the
+        // CLI doesn't know about. Warn, because the output of this command might provide an
+        // incomplete view of the network's config settings.
+        let network_version = rpc.get_version_info().await?.protocol_version;
+        let self_version = Version::parse(stellar_xdr::VERSION.pkg)?.major;
+        if self_version < network_version.into() {
+            print.warnln(format!("Network protocol version is {network_version} but the stellar-cli supports {self_version}. The config fetched may not represent the complete config settings for the network. Upgrade the stellar-cli."));
+        }
+
+        // Collect the ledger entries for all the config settings.
+        let keys = ConfigSettingId::variants().map(|id| {
```

Comment:
TIL these are ledger keys (I figured they were but had never seen an example); nice!

Reply:
Yeah it's pretty convenient the config is so accessible just like any other ledger entry. Glad it wasn't some unique construction stored somewhere else.